src/mongocxx/database.hpp

```diff
@@ -248,6 +248,8 @@ class MONGOCXX_API database {
 
     class MONGOCXX_PRIVATE impl;
     std::unique_ptr<impl> _impl;
+    const impl& get_impl() const;
+    impl* get_impl();
```

Comment:
+1 it is fairly unexpected to have there return different types (modulo-constness)